hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecord.java

```diff
@@ -391,6 +392,11 @@ protected static boolean hasMetaFields(Schema schema) {
     return schema.getField(HoodieRecord.RECORD_KEY_METADATA_FIELD) != null;
   }
 
+  public long estimateSerializedSize() {
+    return ObjectSizeCalculator.getObjectSize(this);
```

Comment:
I think it should be `data`, not the whole object?

Reply:
Actually I think we can get the dataSize differently from `UnsafeRow` and `BaseAvroPayload`, `UnsafeRow` could use `getSizeInBytes` to get data size, while `BaseAvroPayload` has `recordBytes` field, can extract the size.

Reply:
@boneanxs thanks boneanxs !  I'll optimize this code

Reply:
@boneanxs  pl review again, thanks boneanxs

Reply:
We can overwrite `HoodieAvroRecord` and `HoodieSparkRecord` to provide bytes size directly.

```java
// HoodieSparkRecord
  @Override
  public long estimateSerializedDataSize() {
    if (data instanceof UnsafeRow) {
      return ((UnsafeRow) data).getSizeInBytes();
    }

    return super.estimateSerializedDataSize();
  }
```

```java
// HoodieAvroRecord
  @Override
  public long estimateSerializedDataSize() {
    if (data instanceof BaseAvroPayload) {
      return ((BaseAvroPayload)data).recordBytes.length;
    }

    return super.estimateSerializedDataSize();
  }
```